### DISCUSSION TITLE
Convert color name to HEX

### DISCUSSION THREAD

**Discussion by ashahabov** [mono/SkiaSharp#2250] [Q&A] [answered]
Can SkiaSharp convert the color name to the corresponding HEX?

By `System.Drawing` I do this in the following way:

```c#
var color = Color.FromName("blue");
var argb = color.ToArgb();
var colorHex = $"{argb:x6}";
```

but I can't find similar in SkiaSharp.

**Comment by mattleibow** [comment]
ToString maybe? Or just cast to uint.

**Reply by ashahabov** [reply]
@mattleibow, thank you for the reply.

I can create `SKColor` instance from HEX string and get HEX of `SKColor`

```c#
var color = SKColor.Parse("#FF0000FF");
var hex = color.ToString();
```

but how to create `SKColor` from color name? E.g, the following doesn't work:

```c#
var color = SKColor.Parse("blue");
```

It expects only HEX string.

---

My input is string color name, which I need to convert to HEX.

**Reply by mattleibow** [reply]
What happens if you do "Blue"? I wonder if the name code is case sensitive by accident... If that is the case then we need to fix that.

**Reply by mattleibow** [reply]
Just looked at the implementation and I see I do not have any code to do name parsing... A pity. I think we should fix this.

Can you open an issue for this?

**Reply by ashahabov** [reply]
Ok. I will.

**Reply by mattleibow** [reply]
thanks! I'll see what we can do.

**Comment by ashahabov** [comment, marked as answer]
Found solution via reflection:

```c#
var inputClrName = "blue";

var fieldInfo = typeof(SKColors).GetFields(BindingFlags.Static | BindingFlags.Public)
                                             .First(fieldInfo => fieldInfo.Name.ToLower() == inputClrName);
var color = (SKColor)fieldInfo.GetValue(null);

var outputHexString = color.ToString();
```

**Reply by mattleibow** [reply]
I think we can do better. Can you open an issue and we can fix this for real?
